packages/dev/core/src/Meshes/mesh.ts

```diff
@@ -3000,88 +3000,26 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
     }
 
     /**
-     * Modify the mesh to get a flat shading rendering.
-     * This means each mesh facet will then have its own normals. Usually new vertices are added in the mesh geometry to get this result.
-     * Warning : the mesh is really modified even if not set originally as updatable and, under the hood, a new VertexBuffer is allocated.
-     * @returns current mesh
+     * @internal
+     * @param indices 
+     * @param positions 
+     * @returns the flattened Normal data
      */
-    public convertToFlatShadedMesh(): Mesh {
-        const kinds = this.getVerticesDataKinds();
-        const vbs: { [key: string]: VertexBuffer } = {};
-        const data: { [key: string]: FloatArray } = {};
-        const newdata: { [key: string]: Array<number> } = {};
-        let updatableNormals = false;
-        let kindIndex: number;
-        let kind: string;
-
-        for (kindIndex = 0; kindIndex < kinds.length; kindIndex++) {
-            kind = kinds[kindIndex];
-            const vertexBuffer = <VertexBuffer>this.getVertexBuffer(kind);
-
-            // Check data consistency
-            const vertexData = vertexBuffer.getData();
-            if (vertexData instanceof Array || vertexData instanceof Float32Array) {
-                if (vertexData.length === 0) {
-                    continue;
-                }
-            }
+    private _getFlattenedNormals(indices: IndicesArray, positions: FloatArray): Float32Array {
+        const normals = new Float32Array(indices.length * 3);
+        let normalsCount = 0;
 
-            if (kind === VertexBuffer.NormalKind) {
-                updatableNormals = vertexBuffer.isUpdatable();
-                kinds.splice(kindIndex, 1);
-                kindIndex--;
-                continue;
-            }
+        // Decide if normals should be flipped
+        const flipNormalGeneration = this._scene.useRightHandedSystem
+            ? this.overrideMaterialSideOrientation === Constants.MATERIAL_CounterClockWiseSideOrientation
+            : this.overrideMaterialSideOrientation === Constants.MATERIAL_ClockWiseSideOrientation;

```

Comment:
I know I helped you write this part, but it could be further simplified:
```suggestion
        const flipNormalGeneration = this.overrideMaterialSideOrientation === (this._scene.useRightHandedSystem ? Constants.MATERIAL_CounterClockWiseSideOrientation : Constants.MATERIAL_ClockWiseSideOrientation);
```

Reply:
Sure!